src/api/RestAPI.ts

```diff
@@ -336,25 +337,37 @@ export class RestAPI extends EventEmitter {
          */
         iterationCallback?: (response: Response) => void
     ): Promise<Response> {
-        return execRequest(async () => {
-            const repeater = async (): Promise<Response> => {
-                const result = await promise();
-                iterationCallback?.(result);
-
-                if (cancelCondition?.(result)) {
-                    return null;
-                }
-
-                if (!successCondition(result)) {
-                    await new Promise((resolve) => setTimeout(resolve, interval)); // sleep to avoid firing requests too fast
-                    return repeater();
+        let internalErrrorCount = 0;
+        while (internalErrrorCount <= MAX_INTERNAL_ERROR_RETRY) {
```

Comment:
You likely do not need the `while` here if you move the try/catch block around `const result = await promise();` only which would make the logic easier to read as you wont need to follow the returns in the loop.